dataland-community-manager/src/main/kotlin/org/dataland/datalandcommunitymanager/services/EmailBuilder.kt

```diff
@@ -0,0 +1,154 @@
+    @Value("\${dataland.notification.receivers.cc.bulk.data.request}")
+    private val propNotificationReceiversCcBulkDataRequest: String,
+) {
+    private val datalandNotificatorEmailContact =
+        EmailContact(assertEmailAddressFormatAndReturnIt(propNotificationSenderAddress), propNotificationSenderName)
+
+    private val notificationReceiversBulkDataRequest =
+        getEmailContactsFromProp(propNotificationReceiversBulkDataRequest)
+    private val notificationReceiversCcBulkDataRequest =
+        getEmailContactsFromProp(propNotificationReceiversCcBulkDataRequest)
+
+    private val bulkDataRequestNotificationMailStyle =
+        """
+    <style>
+    body {
+                    font-family: Arial, sans-serif;
+                    color: #333;
+                }
+                .container {
+                    max-width: 600px;
+                    margin: 0 auto;
+                    padding: 20px;
+                    border-radius: 10px;
+                }
+                .header {
+                    font-size: 24px;
+                    font-weight: bold;
+                    margin-bottom: 10px;
+                }
+                .section {
+                    margin-bottom: 10px;
+                }
+                .bold {
+                    font-weight: bold;
+                }
+    </style>
+    """
+
+    private fun assertEmailAddressFormatAndReturnIt(emailAddress: String): String {
+        val regexForValidEmail = Regex("^[a-zA-Z0-9_.!-]+@[a-zA-Z0-9-]+.[a-z]{2,3}\$")
+        if (!regexForValidEmail.matches(emailAddress)) {
+            val logger = LoggerFactory.getLogger(javaClass)
+            logger.error(
+                "The email addresses provided by the Spring properties have a wrong format. " +
+                    "The following email address was parsed from that prop and caused this error: $emailAddress" +
+                    "The Spring application is shutting down because sending notifications might not work as expected.",
+            )
+            exitProcess(1)
+        }
+        return emailAddress
+    }
+
+    private fun getEmailContactsFromProp(propWithSemicolonSeperatedEmailAddresses: String): List<EmailContact> {
+        return propWithSemicolonSeperatedEmailAddresses.split(";").map {
+                emailAddressString ->
+            EmailContact(assertEmailAddressFormatAndReturnIt(emailAddressString))
+        }
+    }
+
+    private fun buildUserInfo(): String {
+        val user = DatalandAuthentication.fromContext() as DatalandJwtAuthentication
+        return "User ${user.username} (Keycloak id: ${user.userId})"
+    }
+
+    private fun buildBulkDataRequestEmailText(
+        bulkDataRequest: BulkDataRequest,
+        acceptedCompanyIdentifiers: List<String>,
+    ): String {
+        return "A bulk data request has been submitted: " +
+            "Environment: $propProxyPrimaryUrl " +
+            "User: ${buildUserInfo()} " +
```

Comment:
Is the information who requested the data necessary?
Also, is this fine from a data privacy point of view? We now send emails containing the username (email address) to the recipients of the receiver list. So once a user once their account to be deleted the information about their user accounts indirectly lives on in someones email account.

Reply:
Closed as this was moved to a task https://jira.d-fine.dev/browse/DALA-2773